OpenSim/Common/DataTable.h

```diff
@@ -843,8 +843,9 @@ class DataTable_ : public AbstractDataTable {
         const size_t totalWidth{std::accumulate(columnWidths.cbegin(),
                                                 columnWidths.cend(),
                                                 static_cast<size_t>(0))};
-        result.reserve((totalWidth * table.size() * excessAllocation) +
-                       result.capacity());
+        result.reserve(result.capacity() +
+            static_cast<unsigned>(std::roundf(
+                totalWidth * table.size() * excessAllocation)));
```

Comment:
Actually, I have two concerns.

1. It isn't clear why `excessAllocation` has been set to 1.25 specifically. A comment on L762 or L845 would be helpful to explain the intention, as it seems like the most natural action (which would not require a comment) would be to allocate only as much memory as is required.

2. Because it isn't clear why `excessAllocation` has been set to 1.25, it therefore isn't immediately obvious that truncation via (implicit or explicit) casting is appropriate. I think I've used a language at some point that didn't allow casting to an int without floor/ceil/round first, so I'm probably biased to see something like `static_cast<unsigned>(3.14)` as a bug.